STEP_Project/OUListContent.js

```diff
@@ -16,48 +49,69 @@ function initButtons(tabl)
     let orgUnits = tabl.rows;
     let numOrgUnits = orgUnits.length;
     let numRows = numOrgUnits-1;
-    addButtons(orgUnits,numRows);
-
+    
     // This is to handle an edge case:
     // It deals with the last row in the table that doesn't actually correspond to an OU
-    
     let row = orgUnits[numOrgUnits-1];
     const button = document.createElement('p');
-    button.setAttribute('class','bClass');
+    button.setAttribute('class','pClass');
     row.appendChild(button);
+
+    addButtons();
 }
 
-/**
- * This adds a button to every row in the table
- * If the row already has a button, it skips that row
- * @param {HTML Object} OUs - List of OUs
- * @param {integer} numRows - Number of rows
- */
-function addButtons(orgUnits, numRows)
+function addButtonsToRows(data)
 {
+    let tabl = document.querySelector('table[role=grid]');
+    let orgUnits = tabl.rows;
+    let numOUs = orgUnits.length;
+    let numRows = numOUs;
+  
     //skip the first row as that's the heading row and does not represent an OU.
     for (let i = 1; i < numRows; i++) 
     {
         let row = orgUnits[i];
-        let temp = row.getElementsByClassName('bClass');
-        if (temp.length == 0)
+        let temp = row.getElementsByClassName('pClass');
+        let temp2 = row.getElementsByClassName('mClass');
+        if (temp.length == 0 && temp2.length==0)
         {
             const button = document.createElement('td');
-            button.innerHTML = 'click me';
-            button.setAttribute('class','bClass');
+            let dataid = row.getAttribute("data-row-id");
+            let dataname = data[dataid]
+            button.setAttribute('class','pClass');
+            if (dataname!=undefined)
+            {
+                button.innerHTML = "-";
+                button.setAttribute('class','mClass');
+            }
+            else
+            {
+                button.innerHTML="+";
+            }        
             row = orgUnits[i];
             row.appendChild(button);
         }
     }
 }
 
+/**
+ * This adds a button to every row in the table
+ * If the row already has a button, it skips that row
+ */
+function addButtons()
+{
+    OBJ.get(null, addButtonsToRows);
```

Comment:
Does setting null here return all data in chrome storage ? I think it'll make sense that the OBJ or whatever it is renamed to, is directly used to check if the corresponding data-row-id is present or not. Passing on all of the storage data might be a heavy operation.

Reply:
We actually have to check for every single row. The options are:

- Pass the entire storage at once, and loop through the OUs. 

- Pass the storage entries, one OU at a time.

Since rendering the storage is asynchronous, I think it would be more expensive to pass each individual OU one by one, because in the end we have to pass all OUs. So I decided to pass them all together, so that the rendering from storage happens once instead of every time.